quickwit/quickwit-storage/src/object_storage/s3_compatible_storage.rs

```diff
@@ -554,6 +565,96 @@ impl Storage for S3CompatibleObjectStorage {
         Ok(())
     }
 
+    async fn bulk_delete<'a>(&self, paths: &[&'a Path]) -> Result<(), BulkDeleteError> {
+        let mut error = None;
+        let mut successes = Vec::with_capacity(paths.len());
+        let mut failures = HashMap::new();
+        let mut unattempted = Vec::new();
+
+        #[cfg(test)]
+        const MAX_NUM_KEYS: usize = 3;
+
+        #[cfg(not(test))]
+        const MAX_NUM_KEYS: usize = 1_000;
+
+        for chunk in paths.chunks(MAX_NUM_KEYS) {
+            if error.is_some() {
+                unattempted.extend(chunk.iter().map(|path| path.to_path_buf()));
+                continue;
+            }
+            let objects: Vec<ObjectIdentifier> = chunk
+                .iter()
+                .map(|path| ObjectIdentifier {
+                    key: self.key(path),
+                    ..Default::default()
+                })
+                .collect();
+            let delete = Delete {
+                objects,
+                ..Default::default()
+            };
+            let delete_objects_req = DeleteObjectsRequest {
+                bucket: self.bucket.clone(),
+                delete,
+                ..Default::default()
+            };
+            let delete_objects_res = retry(&self.retry_params, || async {
+                self.s3_client
+                    .delete_objects(delete_objects_req.clone())
+                    .await
+                    .map_err(RusotoErrorWrapper::from)
+            })
+            .await;
+
+            match delete_objects_res {
+                Ok(delete_objects_output) => {
+                    if let Some(deleted_objects) = delete_objects_output.deleted {
+                        for deleted_object in deleted_objects {
+                            if let Some(key) = deleted_object.key {
+                                let path = self.relative_path(&key);
+                                successes.push(path);
+                            }
+                        }
+                    }
+                    if let Some(s3_errors) = delete_objects_output.errors {
+                        for s3_error in s3_errors {
+                            if let Some(key) = s3_error.key {
+                                let path = self.relative_path(&key);
+                                match s3_error.code {
+                                    Some(code) if code == "NoSuchKey" => {
```

Comment:
Ok so we consider that deleting a non existing object is a success, right?
This can be a bit misleading. Can you indicate in the docs that successes contains non existing object?
Maybe we can log something only?